server/web/server.go

```diff
@@ -182,7 +199,22 @@ func (app *HttpServer) Run(addr string, mws ...MiddleWare) {
 				if app.Cfg.Listen.ListenTCP4 {
 					server.Network = "tcp4"
 				}
-				if err := server.ListenAndServe(); err != nil {
+				ln, err := net.Listen(server.Network, app.Server.Addr)
```

Comment:
if you want to support the callbacks, I think you need to think about `app.Cfg.Listen.EnableFcgi` and `app.Cfg.Listen.EnableHTTPS || app.Cfg.Listen.EnableMutualHTTPS`, all cases.